rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/transform/TbCopyKeysNodeTest.java

```diff
@@ -156,6 +161,20 @@ void givenMsgDataNotJSONObject_whenOnMsg_thenTVerifyOutput() throws Exception {
         assertThat(newMsg).isSameAs(msg);
     }
 
+    @Test
+    void givenOldConfig_whenUpgrade_thenShouldReturnTrueResultWithNewConfig() throws Exception {
+        // GIVEN
+        var config = new TbCopyKeysNodeConfiguration().defaultConfiguration();
+        var oldConfigJson = (ObjectNode) JacksonUtil.valueToTree(config);
+        oldConfigJson.remove("copyFrom");
+        oldConfigJson.put("fromMetadata", "false");
+        // WHEN
+        TbPair<Boolean, JsonNode> upgrade = node.upgrade(0, oldConfigJson);
+        // THEN
+        assertTrue(upgrade.getFirst());
+        assertEquals(config, JacksonUtil.treeToValue(upgrade.getSecond(), config.getClass()));
+    }
```

Comment:
i think it is better to manually provide json string for both cases and set up parameterized test with two cases

Reply:
same in delete node test